core/src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -445,6 +445,10 @@ public void randomIndexTemplate() throws IOException {
                 mappings.endObject().endObject();
             }
 
+            for (String setting : randomSettingsBuilder.internalMap().keySet()) {
+                assertThat("non index. prefix setting set on index template, its a node setting...", setting, startsWith("index."));
```

Comment:
yea, its a different issue, agreed. Another solution might be full validation of index settings allowed with a flag for dynamic ones, similar to what we do with dynamic settings.